### PR TITLE
CSES Increasing Subsequences II Revision II

### PR DESCRIPTION
**Pull request by timothygao8710**
If someone could tell me how to make edits in a pull request that would be great lol

### CODE REVIEW

**Comment by frodakcin** [comment]
If you push to the branch or repository where you performed the initial pull request, it will automatically update the PR

**Comment by frodakcin** [comment]
Is it fine to close https://github.com/cpinitiative/usaco-guide/pull/593 ?

**Comment by frodakcin** [comment]
lol is https://github.com/cpinitiative/usaco-guide/pull/598 also from the same update chain?

**Comment by timothygao8710** [comment]
Yea lol I'm pretty new to github. Feel free to close the duplicates.

**Comment by frodakcin** [comment]
no worries

File: solutions/CSES Increasing Subsequence II.mdx [hunk 1]
```diff
@@ -2,10 +2,24 @@
...
 ---
 
-Coordinate compress, then range sum.
+Let $dp[i]$ be the number of increasing subsequences of $x$ ending on the right at index $i$.
+Notice that $dp[i]$ can be calculated as the summation of all $dp[j]$ such that $j < i$ and $x_j < x_i$.
+
+Naively, this gives an $\mathcal O(N^2)$ solution. In order to speed this up, we can
+coordinate compress, then range sum.
+
+More specifically, we sort all numbers in $x$ and map each distinct number to its index in the sorted array. This way, we effectively coordinate compress the numbers.
```
Comment (dolphingarlic) [inline solutions/CSES Increasing Subsequence II.mdx:14]:
```suggestion
More specifically, we sort all numbers in $x$ and map each distinct number to its index in the sorted array. This is known as coordinate compression, and it makes the range of the numbers significantly smaller.
```

File: solutions/CSES Increasing Subsequence II.mdx [hunk 2]
```diff
@@ -2,10 +2,24 @@
 id: cses-1748
 source: CSES
 title: Increasing Subsequence II
-author: Benjamin Qi, Óscar Garries
+author: Benjamin Qi, Óscar Garries, Timothy Gao
 ---
 
-Coordinate compress, then range sum.
+Let $dp[i]$ be the number of increasing subsequences of $x$ ending on the right at index $i$.
```
Comment (dolphingarlic) [inline solutions/CSES Increasing Subsequence II.mdx:8]:
```suggestion
Let $dp[i]$ be the number of increasing subsequences of $x$ ending on the right of index $i$.
```

File: solutions/CSES Increasing Subsequence II.mdx [hunk 3]
```diff
@@ -2,10 +2,24 @@
...
+Notice that $dp[i]$ can be calculated as the summation of all $dp[j]$ such that $j < i$ and $x_j < x_i$.
```
Comment (dolphingarlic) [inline solutions/CSES Increasing Subsequence II.mdx:9]:
```suggestion
Notice that $dp[i]$ can be calculated as the summation of all $dp[j]$ where $j < i$ and $x_j < x_i$.
```

File: solutions/CSES Increasing Subsequence II.mdx [hunk 4]
```diff
@@ -2,10 +2,24 @@
...
+
+Naively, this gives an $\mathcal O(N^2)$ solution. In order to speed this up, we can
+coordinate compress, then range sum.
+
+More specifically, we sort all numbers in $x$ and map each distinct number to its index in the sorted array. This way, we effectively coordinate compress the numbers. 
+
+With this, we can create a [BIT](https://usaco.guide/gold/PURS), which intially starts with all 0s. As we run through $x$ from left to right, $dp[i]$ would be summation of all $j<k$ in the BIT, where $k$ is the index $x_i$ maps to. Then, we update $k$ in the BIT as preparation for the next query by another number. 
+
+**Time Complexity:** $\mathcal{O}(NlogN)$
```
Comment (dolphingarlic) [inline solutions/CSES Increasing Subsequence II.mdx:18]: Move this line to the top of the solution

File: solutions/CSES Increasing Subsequence II.mdx [hunk 5]
```diff
@@ -2,10 +2,24 @@
...
-Coordinate compress, then range sum.
+Let $dp[i]$ be the number of increasing subsequences of $x$ ending on the right at index $i$.
+Notice that $dp[i]$ can be calculated as the summation of all $dp[j]$ such that $j < i$ and $x_j < x_i$.
+
+Naively, this gives an $\mathcal O(N^2)$ solution. In order to speed this up, we can
+coordinate compress, then range sum.
+
+More specifically, we sort all numbers in $x$ and map each distinct number to its index in the sorted array. This way, we effectively coordinate compress the numbers. 
+
+With this, we can create a [BIT](https://usaco.guide/gold/PURS), which intially starts with all 0s. As we run through $x$ from left to right, $dp[i]$ would be summation of all $j<k$ in the BIT, where $k$ is the index $x_i$ maps to. Then, we update $k$ in the BIT as preparation for the next query by another number.
```
Comment (dolphingarlic) [inline solutions/CSES Increasing Subsequence II.mdx:16]:
```suggestion
With this, we can create a [BIT](https://usaco.guide/gold/PURS), where each value is initially 0. As we iterate through $x$, $dp[i]$ would be the sum of the values at positions $j<k$ in the BIT, where $k$ is the index that $x_i$ maps to. We then increase position $k$ in the BIT by $dp[i]$ as preparation for the next element of $x$. 
```

**Review by dolphingarlic — CHANGES_REQUESTED** [review]
Just some grammar/style corrections and then you should be good to go

**Pull request merged**